packages/sit-onyx/src/components/OnyxSwitch/OnyxSwitch.ct.tsx

```diff
@@ -1,48 +1,166 @@
+import { createScreenshotsForAllStates } from "../../utils/playwright";
 import { expect, test } from "../../playwright-axe";
 import OnyxSwitch from "./OnyxSwitch.vue";
 
-test("should render default OnyxSwitch", async ({ mount, makeAxeBuilder }) => {
+test("should render unchecked", async ({ mount, makeAxeBuilder }) => {
   // ARRANGE
-  const component = await mount(<OnyxSwitch label="Switch" />);
-
-  // ASSERT
-  await expect(component).toContainText("Switch");
-  await expect(component).toHaveScreenshot("default.png");
+  const component = await mount(<OnyxSwitch label="Unchecked" />);
 
   // ACT
   const accessibilityScanResults = await makeAxeBuilder().analyze();
 
   // ASSERT
   expect(accessibilityScanResults.violations).toEqual([]);
-});
 
-test("should render disabled OnyxSwitch", async ({ mount, makeAxeBuilder }) => {
-  // ARRANGE
-  const component = await mount(<OnyxSwitch label="Switch" disabled />);
+  const onyxSwitch = component.getByRole("checkbox");
+
+  // ACT
+  await onyxSwitch.hover();
 
   // ASSERT
-  await expect(component).toContainText("Switch");
-  await expect(component).toBeDisabled();
-  await expect(component).toHaveScreenshot("disabled.png");
+  await expect(onyxSwitch).not.toBeChecked();
+});
+
+test("should render checked", async ({ mount, makeAxeBuilder }) => {
+  // ARRANGE
+  const component = await mount(<OnyxSwitch label="Checked" modelValue={true} />);
 
   // ACT
   const accessibilityScanResults = await makeAxeBuilder().analyze();
 
   // ASSERT
   expect(accessibilityScanResults.violations).toEqual([]);
+
+  const onyxSwitch = component.getByRole("checkbox");
+
+  // ACT
+  await onyxSwitch.hover();
+
+  // ASSERT
+  await expect(onyxSwitch).toBeChecked();
 });
 
-test("should render invalid OnyxSwitch", async ({ mount, makeAxeBuilder }) => {
+test("should render required", async ({ mount, makeAxeBuilder }) => {
```

Comment:
This doesn't really test anything :) Imho its fine if we remove it and just have the screenshot